File: src/main/java/ru/HuffmanNode.java
package ru;

import java.io.IOException;
import java.io.InputStream;

public class HuffmanNode {
    public int leftNodeId;
    public int rightNodeId;

    public HuffmanNode(InputStream r) throws IOException {
        this.leftNodeId = TlkFile.readInt32(r);
        this.rightNodeId = TlkFile.readInt32(r);
    }
}

File: python_ver/huffman_node.py
import input_stream


class HuffmanNode:
    def __init__(self, stream):
        left_val = stream.read_int_32()
        right_val = stream.read_int_32()
        if left_val >= 2 ** 31:
            left_val -= 2 ** 32
        if right_val >= 2 ** 31:
            right_val -= 2 ** 32
        self.left_node_id = left_val
        self.right_node_id = right_val
